src/ompl/base/PlannerTerminationCondition.h

```diff
@@ -120,6 +121,32 @@ namespace ompl
 
         /** \brief Return a termination condition that will become true \e duration seconds in the future (wall-time), but is checked in a separate thread, every \e interval seconds; \e interval must be less than \e duration */
         PlannerTerminationCondition timedPlannerTerminationCondition(double duration, double interval);
+
+        /** \brief Return a termination condition that will become true as soon as the problem definition has an exact solution */
+        PlannerTerminationCondition exactSolnPlannerTerminationCondition(ompl::base::ProblemDefinitionPtr pdef);
+
+        /** \brief A class to run a planner for a specific number of iterations. Casts to a PTC for use with Planner::solve */
+        class IterationTerminationCondition
```

Comment:
Just chiming in on this: I completely agree with Mark about the risk of people trying to compare iteration counts of algorithms, but in every OMPL project I've worked on I've needed to implement my own iteration-based PTC. The main reason is because an iteration-based, non-threaded PTC enables deterministic planner output given a particular random seed. This is extremely useful for testing.